airbyte-integrations/connectors/source-mysql/src/main/java/io/airbyte/integrations/source/mysql/MySqlSource.java

```diff
@@ -145,22 +150,39 @@ public JsonNode toDatabaseConfig(final JsonNode config) {
     // in the setJsonField method in MySqlSourceOperations.java
     jdbcUrl.append("&yearIsDateType=true");
     if (config.get(JdbcUtils.JDBC_URL_PARAMS_KEY) != null && !config.get(JdbcUtils.JDBC_URL_PARAMS_KEY).asText().isEmpty()) {
-      jdbcUrl.append("&").append(config.get(JdbcUtils.JDBC_URL_PARAMS_KEY).asText());
+      jdbcUrl.append(JdbcUtils.AMPERSAND).append(config.get(JdbcUtils.JDBC_URL_PARAMS_KEY).asText());
     }
-
+    Map<String, String> additionalParameters = new HashMap<>();
     // assume ssl if not explicitly mentioned.
     if (!config.has(JdbcUtils.SSL_KEY) || config.get(JdbcUtils.SSL_KEY).asBoolean()) {
-      jdbcUrl.append("&").append(String.join("&", SSL_PARAMETERS));
+      if (config.has(JdbcUtils.SSL_MODE_KEY)) {
+        if (DISABLE.equals(config.get(JdbcUtils.SSL_MODE_KEY).get(JdbcUtils.MODE_KEY).asText())) {
+          jdbcUrl.append(JdbcUtils.AMPERSAND).append("sslMode=DISABLE");
+        } else {
+          additionalParameters.putAll(obtainConnectionOptions(config.get(JdbcUtils.SSL_MODE_KEY)));
```

Comment:
As for me better to avoid import static method and use MySqlSslConnectionUtils.obtainConnectionOptions, wdyt?

Reply:
MySqlSslConnectionUtils is a utility class, so I'll leave the method you specified static.